pinot-common/src/main/java/org/apache/pinot/common/utils/config/QueryOptionsUtils.java

```diff
@@ -145,6 +149,33 @@ public static boolean isSkipScanFilterReorder(Map<String, String> queryOptions)
     return "false".equalsIgnoreCase(queryOptions.get(QueryOptionKey.USE_SCAN_REORDER_OPTIMIZATION));
   }
 
+  @Nullable
+  public static Map<String, Set<FieldConfig.IndexType>> getIndexSkipConfig(Map<String, String> queryOptions) {
+    // Example config:  indexSkipConfig='col1=inverted,range&col2=inverted'
+    String indexSkipConfigStr = queryOptions.get(QueryOptionKey.INDEX_SKIP_CONFIG);
+    if (indexSkipConfigStr == null) {
+      return null;
+    }
+
+    String[] perColumnIndexSkip = indexSkipConfigStr.split("&");
+    Map<String, Set<FieldConfig.IndexType>> indexSkipConfig = new HashMap<>();
+    for (String columnConf : perColumnIndexSkip) {
+      String[] conf = columnConf.split("=");
+      if (conf.length != 2) {
+        continue;
```

Comment:
Shouldn't we fail in this case?

Reply:
I think so, ya.  I feel like failed parsing is better to end up as a failed query than a successful query without the intended options set.

Reply:
Actually it looks like there isn't precedent for throwing exceptions from within `InstancePlanMakerImplV2#applyQueryOptions`, so the "easiest" would be a `RuntimeException`